### ISSUE TITLE
What are helpful Keyboard Shortcuts for the Ruler Interaction?

### ISSUE THREAD

**Issue by terracoda** [phetsims/gravity-force-lab#134]
@arouinfar  and @emily-phet, on Tuesday Dec 17 @mbarlow12 and I discussed possible shortcuts that could be useful for moving the ruler in GFL.

These are the 3 that I came up with that would work the same as the Jump keys in BASE. The J key is semi-modal key (like a Shift key) and then a second letter is added to jump to the intended place:
- [J] + [S]: where S represents sphere. A shortcut combination to measure distance between spheres. Pressing J+S would place the zero mark of ruler at center of the m1 sphere, and an alert read out the distance between the spheres.
- [J] + [H]: where H represents Home position. A shortcut combination to return to the ruler to its home position just below Sphere position sliders. The user could then press Space, Enter or Esc, or Tab to release the ruler.

I also proposed:
- [J] + [V]: where V represents vector. A shortcut combination to jump to the start of a force vector and read out the length of the vector arrow.  Pressing J+V would place the zero mark of the ruler on the start of m1’s force vector arrow and read out it’s length. If vector is off screen, we might need a special alert, “off screen, too huge to measure”.

@mbarlow12 pointed out that the actual length of the vector arrows might not be a value that any one needs, so I am reaching out to you both for feedback on these ideas fro shortcuts. @mbarlow12 and I both like the J+S and J+H shortcuts.

**My main questions are**
1. Is a J+V shortcut at all useful?
2. Do you agree that J+S and J+H would be useful for some students?

**If yes to question 2**
-  I will need to create some alerts, and 
- we will eventually need some visual additions for the Keyboard Shortcuts dialog.

Other things like, snap-to behaviour, default step-size, and larger step size will be worked out later, but I thought it would useful to get feedback on these potential shortcut combinations.

Here's the link to [the PDOM section in design document ](https://docs.google.com/document/d/1-37qAgde2XrlXBQae2SgjartM35_EnzDD9pdtd3nXAM/edit?ts=5c12c712#heading=h.ea6pw1gdk0s5) if you would like to see more details in context of the grab button and grab alerts.

@emily-phet, as previously discussed, I do understand that the accessibility and usability of the ruler is our last priority in terms of our design and implementation flow.

**Comment by arouinfar** [comment]
> [J] + [S]: where S represents sphere. A shortcut combination to measure distance between spheres. Pressing J+S would place the zero mark of ruler at center of the m1 sphere, and an alert read out the distance between the spheres.

Love this idea! We will need to change the bounds of the ruler so that it's possible to place the 0 mark on m1 when it's on the far right side of the play area. Currently, this is the limit:

<img width="696" alt="screen shot 2018-12-20 at 10 29 25 am" src="https://user-images.githubusercontent.com/8419308/50300693-6cba1d80-0442-11e9-8989-2889da0e99be.png">

> [J] + [H]: where H represents Home position. A shortcut combination to return to the ruler to its home position just below Sphere position sliders. The user could then press Space, Enter or Esc, or Tab to release the ruler.

Great idea! I'd be in favor of this change.

> [J] + [V]: where V represents vector. A shortcut combination to jump to the start of a force vector and read out the length of the vector arrow. Pressing J+V would place the zero mark of the ruler on the start of m1’s force vector arrow and read out it’s length. If vector is off screen, we might need a special alert, “off screen, too huge to measure”.

The vector lengths are meant to be more qualitative, and do not scale linearly due to the large range of possible values. Providing this short cut may encourage students to try and make quantitative sense of a qualitative representation, so I would not be in favor of this one.

**Comment by emily-phet** [comment]
@terracoda I agree with @arouinfar re J+V (let's not) and J+H (let's do this one). For the J+S, do the descriptions in GFL and GFLB refer to them as a sphere, or mass? We should use whichever one is the word we use most to indicate the sphere/mass thing.

**Comment by terracoda** [comment]
Great, thanks for the enthusiastic feedback. @mbarlow12 and I are in agreement that the vectors are not something that should be measured quantitatively with the ruler.

@emily-phet, regarding wording, we do use the word, "Sphere" in the scene summary, when describing the objects ("m1, Blue Sphere", "m2, Red Sphere"), and when describing position ("Sphere Position" heading). Direct references to the spheres in alerts; however, are always m1 or m2, and in Basics it will likely be Mass 1 and Mass 2. 

That said, J+M is not a good shortcut due to the close proximity of these 2 letters on a qwerty keyboard. 

I think S is a good alternative to M, in this case. I feel there are enough references in the PDOM that mapping S to "Sphere" should not be too difficult. I would have preferred to use M, but operationally, it is not a good shortcut. S and J are both on the home row and J usually has a tactile mark for orientation.

@emily-phet, are you comfortable with moving forward with J+S?

**Comment by terracoda** [comment]
I'll close this issue as I created a new one for implementation.
